shuup/front/middleware.py

```diff
@@ -117,7 +118,10 @@ def _set_basket(self, request):
     def _set_timezone(self, request):
         if request.person.timezone:
             timezone.activate(request.person.timezone)
-            # TODO: Fallback to request.shop.timezone (and add such field)
+        elif get_supplier(request) and get_supplier(request).timezone:
```

Comment:
Maybe, just in case that get_supplier makes some quick db check and is not cached to request:

```
else:
     supplier = get_supplier(request)
     if supplier and supplier.timezone:
          timezone.timezone.activate(get_supplier(request).timezone)
```